Assets/Scripts/Models/Area/Room.cs

```diff
@@ -15,546 +15,360 @@
-            foreach (Tile t in sourceTile.GetNeighbours())
+            if (tiles.Contains(tile))
             {
-                if (t != null && t.Room != null && (onlyIfOutside == false || t.Room.IsOutsideRoom()))
-                {
-                    world.DeleteRoom(t.Room);
-                }
+                // This tile in not in this room.
+                return;
             }
 
-            ActualFloodFill(sourceTile, null, 0);
+            tile.Room = null;
+            tiles.Remove(tile);
         }
-    }
 
-    public void AssignTile(Tile t)
-    {
-        if (tiles.Contains(t))
+        public void ReturnTilesToOutsideRoom()
         {
-            // This tile already in this room.
-            return;
-        }
+            for (int i = 0; i < tiles.Count; i++)
+            {
+                // Assign to outside.
+                tiles[i].Room = World.Current.RoomManager.OutsideRoom;
+            }
 
-        if (t.Room != null)
-        {
-            // Belongs to some other room.
-            t.Room.tiles.Remove(t);
+            tiles = new List<Tile>();
         }
 
-        t.Room = this;
-        tiles.Add(t);
-    }
-
-    public void ReturnTilesToOutsideRoom()
-    {
-        for (int i = 0; i < tiles.Count; i++)
+        public bool IsOutsideRoom()
         {
-            // Assign to outside.
-            tiles[i].Room = World.Current.GetOutsideRoom();
+            return this == World.Current.RoomManager.OutsideRoom;
         }
 
-        tiles = new List<Tile>();
-    }
-
-    public bool IsOutsideRoom()
-    {
-        return this == World.Current.GetOutsideRoom();
-    }
-
-    public int GetSize()
-    {
-        return tiles.Count();
-    }
-
-    // Changes gas by amount in pressure (atm) per tile, evenly distributed over all gases present
-    // Possibly deprecated. Use MoveGasTo(room, amount)
-    public void ChangeGas(float amount)
-    {
-        if (IsOutsideRoom())
+        // Changes gas by amount in pressure (atm) per tile, evenly distributed over all gases present
+        // Possibly deprecated. Use MoveGasTo(room, amount)
+        public void ChangeGas(float amount)
         {
-            return;
-        }
+            if (IsOutsideRoom())
+            {
+                return;
+            }
 
-        List<string> names = new List<string>(atmosphericGasses.Keys);
-        foreach (string name in names)
-        {
-            float fraction = GetGasFraction(name);
-            ChangeGas(name, amount * fraction);
+            List<string> names = new List<string>(atmosphericGasses.Keys);
+            foreach (string name in names)
+            {
+                float fraction = GetGasFraction(name);
+                ChangeGas(name, amount * fraction);
+            }
         }
-    }
 
-    public Tile[] FindExits()
-    {
-        List<Tile> exits = new List<Tile>();
-        foreach (Tile tile in tiles)
+        public Tile[] FindExits()
```

Comment:
Again, this is how it was, the only reason git sees a difference is because of some weirdness but I didn't really change anything there. However, I have changed it to List as it creates one either way